core/state_processor.go

```diff
@@ -191,9 +206,11 @@ func MakeReceipt(evm *vm.EVM, result *ExecutionResult, statedb *state.StateDB, b
 		receipt.ContractAddress = crypto.CreateAddress(evm.TxContext.Origin, tx.Nonce())
 	}
 
-	// Set the receipt logs and create the bloom filter.
-	receipt.Logs = statedb.GetLogs(tx.Hash(), blockNumber.Uint64(), blockHash)
-	receipt.Bloom = types.CreateBloom(receipt)
+	// Annotate the logs with blockhash and blocknumber.
+	for _, log := range receipt.Logs {
+		log.BlockHash = blockHash
```

Comment:
I think these were needed for the pending logs which we don't support anymore. I don't find any usage for them now.

Reply:
I would like to keep this PR as light as possible, if you feel like we can get rid of log.Blockhash, we should do that in a follow-up pr imo